cli/ballerina-cli-module/src/main/java/org/ballerinalang/cli/module/Pull.java

```
@@ -0,0 +1,226 @@
+                FileOutputStream outputStream = new FileOutputStream(baloPath.toString())) {
+            byte[] buffer = new byte[1024];
+            int count;
+
+            long totalSizeInKB = conn.getContentLengthLong() / 1024;
+            try (ProgressBar progressBar = new ProgressBar(
+                    modulePath + ":" + moduleVersion + " [central.ballerina.io -> home repo] ",
+                    totalSizeInKB, 1000, outStream, ProgressBarStyle.ASCII, " KB", 1)) {
+                while ((count = inputStream.read(buffer)) > 0) {
+                    outputStream.write(buffer, 0, count);
+                    progressBar.step();
+                }
+            }
+        } catch (IOException e) {
+            createError("error occurred copying the balo file: " + e.getMessage());
+        }
+    }
+
+    private static void createBaloFileDirectory(Path baloCacheWithModulePath) {
+        try {
+            Files.createDirectory(baloCacheWithModulePath);
+        } catch (Exception e) {
+            createError("error creating directory for balo file");
+        }
+    }
+
+    private static void handleErrorResponse(String url, String modulePath, boolean isBuild, HttpURLConnection conn,
+            int statusCode) {
+        try (BufferedReader reader = new BufferedReader(
+                new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()))) {
+            StringBuilder result = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+
+            MapValue payload = (MapValue) JSONParser.parse(result.toString());
+            String errMsg = payload.getStringValue("message");
+
+            if (statusCode == HttpURLConnection.HTTP_NOT_FOUND && isBuild && errMsg
+                    .contains("module not found")) {
+                // To ignore printing the error
+                createError("");
```

why we need to ignore the error?